### DISCUSSION TITLE
Usage examples

### DISCUSSION THREAD

**Discussion by DhivinX** [xeokit/xeokit-sdk#1417] [Ideas]
Hello,

it would be good to have one collective monorepo with various examples of how to use and implement xeokit-bim-viewer and maybe also xeokit-sdk itself.

Examples would include use in:

- Vue
- Nuxt
- React
- Next
- Svelte
- Capacitor/Cordova (mobile apps)
- plain JavaScript (HTML website)

I can take care of it right away, I would just need to create an empty repository, e.g. xeokit-examples

I would also like to create reactive components for specific technologies, e.g. xeokit-vue, xeokit-react, xeokit-nuxt etc. which we import into our project and use, e.g. <xeo-bim-viewer></xeo-bim-viewer> and wrapped basic attributes. But this is an idea for the future, it would be good to start with examples of use.

**Comment by xeolabs** [comment]
Definitely a good idea. We've had a number of inquiries about whether there are components available for integration with Vue, React, and also Angular. Some boilerplate examples for these would be well-received by users I think.

**Reply by DhivinX** [reply]
Cool! So if you can, please create a xeokit-examples repository and give me access. First, I will create usage examples. Then we will think about dedicated reactive components.

**Reply by DhivinX** [reply]
or `xeokit-integration-examples`

**Reply by xeolabs** [reply]
https://github.com/xeokit/xeokit-integration-examples - is public OK?

**Reply by DhivinX** [reply]
Yes, great thanks!

**Comment by paireks** [comment]
One thing I'd add is I would either call it more specific (e.g. xeokit-integration-examples), as there are already multiple example places in various places (examples list in xeokit-sdk, examples list in xeokit-bim-viewer).

**Comment by xeolabs** [comment]
Name candidates so far:

`xeokit-examples`
`xeokit-integration-examples`
`xeokit-bootstraps`
`xeokit-boilerplates`

Any preferences?

**Comment by xeolabs** [comment]
`xeokit-integration-examples` seems pretty good to me..

**Reply by DhivinX** [reply]
Me too
